### PR TITLE
Loki: remove frontend mode code

### PR DESCRIPTION
**Pull request by gabor**
the frontend-mode in the Loki data source is already disabled-by-default. this pull request completely removes that code.
the changes:
1. the feature-flag `lokiBackendMode` was removed
2. lot of code removed from `datasource.ts` and `result_transformer.ts`
3. lot of test-file-code removed, some tests in `datasource.test.ts` had to be adjusted, because now we have to mock things slightly differently.

NOTE: the live-streaming functionality will be handled separately, it was not modified.

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.test.ts [hunk 1]
```diff
@@ -226,264 +147,85 @@ describe('LokiDatasource', () => {
...
-
-        expect(fieldCache.getFieldByName('Line')?.values.get(0)).toBe('hello');
-        expect(dataFrame.meta?.limit).toBe(20);
-        expect(dataFrame.meta?.searchWords).toEqual(['foo']);
-      });
+    it('should not modify expression with no filters', async () => {
+      expect(ds.applyTemplateVariables(query, {}).expr).toBe(DEFAULT_EXPR);
     });
 
-    it('should return custom error message when Loki returns escaping error', async () => {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/datasource.test.ts:378]: We don't do custom error message for incorrectly escaped labels anymore?

**Review by ivanahuckova — COMMENTED** [review]
LGTM, left 1 question/comment on possible regression? But maybe it is done on purpose. :D

File: public/app/plugins/datasource/loki/datasource.test.ts (hunk 1 above)
Comment (gabor) [reply public/app/plugins/datasource/loki/datasource.test.ts:378]: no, you are correct, we are not doing it currently in backend mode, and we should do that. i'll add a separate PR to add that, if that's ok with you.

**Review by ivanahuckova — APPROVED** [review]
image 
😍 😍 😍 😍

**Comment by gabor** [comment]
the mentioned escaping-error-message will be fixed in https://github.com/grafana/grafana/pull/50352

**Pull request merged**